hazelcast/src/test/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/WanReplicationConfigTransformerTest.java

```diff
@@ -0,0 +1,235 @@
+        assertEquals(WanQueueFullBehavior.THROW_EXCEPTION_ONLY_IF_REPLICATION_ACTIVE, transformer.getWanQueueFullBehaviour(2));
+        assertEquals(WanBatchPublisherConfig.DEFAULT_QUEUE_FULL_BEHAVIOUR, transformer.getWanQueueFullBehaviour(3));
+    }
+
+    @Test
+    public void testGetWanAcknowledgeType() {
+        assertEquals(WanAcknowledgeType.ACK_ON_RECEIPT, transformer.getWanAcknowledgeType(0));
+        assertEquals(WanAcknowledgeType.ACK_ON_OPERATION_COMPLETE, transformer.getWanAcknowledgeType(1));
+        assertEquals(WanBatchPublisherConfig.DEFAULT_ACKNOWLEDGE_TYPE, transformer.getWanAcknowledgeType(2));
+    }
+
+    @Test
+    public void testWanConsumerConfigDefault() {
+        WanConsumerConfig expected = new WanConsumerConfig();
+        WanConsumerConfigHolder holder = transformer.toHolder(expected);
+        WanConsumerConfig actual = transformer.toConfig(holder);
+        assertEquals(expected, actual);
+    }
+
+
+    @Test
+    public void testWanConsumerConfigCustom() {
+        WanConsumerConfig expected = new WanConsumerConfig();
+        expected.setClassName(TEST_CLASS_NAME);
+        expected.setProperties(mockProperties());
+        expected.setPersistWanReplicatedData(true);
+
+        WanConsumerConfigHolder holder = transformer.toHolder(expected);
+        WanConsumerConfig actual = transformer.toConfig(holder);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testWanCustomPublisherDefault() {
+        WanCustomPublisherConfig expected = new WanCustomPublisherConfig();
+        WanCustomPublisherConfigHolder holder = transformer.toHolder(expected);
+        WanCustomPublisherConfig actual = transformer.toConfig(holder);
+        assertEquals(expected, actual); // equality is on the fields of interest
+    }
+
+    @Test
+    public void testWanCustomPublisherConfigCustom() {
+        WanCustomPublisherConfig expected = new WanCustomPublisherConfig();
+        expected.setPublisherId("my-publisher-id");
+        expected.setClassName(TEST_CLASS_NAME);
+        expected.setProperties(mockProperties());
+        WanCustomPublisherConfigHolder holder = transformer.toHolder(expected);
+        WanCustomPublisherConfig actual = transformer.toConfig(holder);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testWanBatchPublisherConfigDefault() {
+        WanBatchPublisherConfig expected = new WanBatchPublisherConfig();
+        mockDiscoverProviderConfig(expected);
+        WanBatchPublisherConfigHolder holder = transformer.toHolder(expected);
+        WanBatchPublisherConfig actual = transformer.toConfig(holder);
+        assertWanBatchPublisherEqual(expected, actual);
+    }
+
+    @Test
+    public void testWanBatchPublisherConfigCustom() {
+        WanBatchPublisherConfig expected = new WanBatchPublisherConfig();
+        expected.setClusterName(UUID.randomUUID().toString());
```

Comment:
I think the setters you didn't use are: 

setImplementation
setSnapshotEnabled
setDiscoveryConfig

Reply:
Added: 

- `setSnapshotEnabled`
- `setDiscoveryConfig` (uses new `DiscoverConfig` codec)